Traceback (most recent call last):
  File "/home/users/vcbenedet/FM30k/auto.py", line 268, in <module>
    met = compute_metrics_at_k(save_path_embeddings, modelo_openai_input_path_name, image_to_legend_indices, k=k)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/users/vcbenedet/FM30k/auto.py", line 187, in compute_metrics_at_k
    similarity_matrix = image_embeddings @ text_embeddings.T  # shape: (N_images, N_captions)
                        ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 18.82 GiB. GPU 0 has a total capacity of 23.65 GiB of which 4.01 GiB is free. Including non-PyTorch memory, this process has 19.63 GiB memory in use. Of the allocated memory 380.58 MiB is allocated by PyTorch, and 18.81 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
